implementations/rust/ockam/ockam_command/src/util/mod.rs

```diff
@@ -610,4 +640,39 @@ mod tests {
             }
         }
     }
+
+    #[test]
+    fn test_process_multi_addr() {
+        let lookup = |node: &str, addr: &str| {
+            let mut l = ConfigLookup::new();
+            l.set_node(node, SocketAddr::from_str(addr).unwrap().into());
+            l
+        };
+        let test_cases = vec![
+            (
+                MultiAddr::from_str("/node/n1").unwrap(),
+                lookup("n1", "127.0.0.0:4000"),
+                "/ip4/127.0.0.0/tcp/4000",
+            ),
+            (
+                MultiAddr::from_str("/project/p1").unwrap(),
+                ConfigLookup::new(),
+                "/project/p1",
+            ),
+            (
+                MultiAddr::from_str("/service/s1").unwrap(),
+                lookup("n1", "127.0.0.0:4000"),
+                "/service/s1",
+            ),
+            (
+                MultiAddr::from_str("/project/p1/node/n1/service/echo").unwrap(),
+                lookup("n1", "127.0.0.0:4000"),
+                "/project/p1/ip4/127.0.0.0/tcp/4000/service/echo",
```

Comment:
These tests are looking great! You could also add some cases to test that it returns the expected errors. It will fail in the following cases:
- The MultiAddr protocol is invalid: `/noode/n1`
- The node is not in the lookup

To do that, I'd define the test_cases as follows, so you can do something similar to what we have at the [`extract_address_value` tests](https://github.com/build-trust/ockam/blob/develop/implementations/rust/ockam/ockam_command/src/util/mod.rs#L606):
```rust
(
    MultiAddr::from_str("/service/s1").unwrap(),
    lookup("n1", "127.0.0.0:4000"),
    Ok("/service/s1"),
),
```

Reply:
I have added a case for the second point. 
Regarding `The MultiAddr protocol is invalid: /noode/n1` , in this case it will fail early during calling `unwrap()`
```Rust
MultiAddr::from_str("/noode/s1").unwrap(),
```
I believe this maybe irrelevant to the tested function (`process_multi_addr`). It may also complicates the test function since I will remove all the `unwrap()`s from the test `vector` , then, inside the loop, 
```Rust
for (ma, lookup, expected) in test_cases {
   ...
}
```
I will start checking if unwraping the `MultiAddr` didn't result in error.  What do u think ?

Reply:
My bad, you are right! Since the input is already a `MultiAddr` that error can't be triggered. Covering the other case is enough then! :+1:

Reply:
Great!